gui/package_app

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash -e
+#
+# package GUI application
+#
+
+function convert_qt_translations()
+{
+# Combine the Qt translation files we use into local qt_??.qm files.
+#
+# It is recommended to combine Qts .qm files, this script does that for
+# linux and macos, windeployqt does this for windows.
+# https://doc.qt.io/qt-5/linguist-programmers.html#deploying-translations
+#
+# This script is created from the log of the windows build from windeployqt
+# with Qt 5.12.1.
+# From the log you can see which translation files are used which depends on
+# which Qt modules we use.
+# In our case these are qtbase_*.qm, qtdeclarative_*qm and qtserialport_*.qm.
+#
+# Note with Qt5 the Qt distributed qt_xx.qm files are metacatalogs, and just
+# copying or converting them won't copy the dependencies.
+
+  if [ "${machine}" = "Mac" ]; then
+    resourcedir="${APPDIR}/Contents/Resources"
+
+    # caution, mktemp here is macos specific version
```

Comment:
Historical note: mktemp was actually a Berkeley thing. The GNU coreutils team knew that it was handy enough they include it in coreutils, which is one of the more popular packages in Homebrew. Macs owned by "programmery types" will thus often have a GNU mktemp as both mktemp and gmktemp that are in directories that can be $PATHed before or after /usr/bin, where the native BSD mktemp lives. This basic battern follows for lots of things and it can be a bit of a landmine of "works on mine" type problems.
Fortunately, here we're just using -t which is the same in both GNU and BSD.